scripts/components/api-changes-validator/usage_statemets_renderer.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  NamespaceDefinitions,
+  UsageStatementsGeneratorOutput,
+} from './types.js';
+import { EOL } from 'os';
+
+/**
+ * Renders usage statements.
+ */
+export class UsageStatementsRenderer {
```

Comment:
Do we need to add unit tests for `usage_statements_runderer` and `import_statements_renderer`?

Reply:
I don't think so.
We have a pattern of using higher level (integration) testing strategies in `scripts` folder. Which seems to be sufficient strategy for now.

Combination of 
https://github.com/aws-amplify/amplify-backend/blob/main/scripts/components/api-changes-validator/api_usage_generator.test.ts
and
https://github.com/aws-amplify/amplify-backend/tree/main/scripts/components/api-changes-validator/test-resources/test-projects ran by https://github.com/aws-amplify/amplify-backend/blob/main/scripts/components/api-changes-validator/api_changes_validator.test.ts
provide enough coverage.

The reason for this is that components in `scripts` often rely on external tools and proving that those integrations work is more important than pedantic unit testing (which in this case would be an art to check a box rather than something useful). Also, higher level tests in `scripts` are fast and stable enough to not worry about their volume.

Reply:
We had simialar discussion here https://github.com/aws-amplify/amplify-backend/pull/800#discussion_r1423269820 .